controller/collaborators.go

```diff
@@ -282,9 +299,41 @@ func (c *CollaboratorsController) getPolicy(ctx collaboratorContext, req *goa.Re
 	if err != nil {
 		return nil, nil, goa.ErrNotFound(err.Error())
 	}
-	policy, pat, err := c.policyManager.GetPolicy(ctx, req, policyID)
+	if spaceForMissingResource != nil {
+		policy, pat, err = c.createPolicy(ctx, req, *spaceForMissingResource)
+	} else {
+		policy, pat, err = c.policyManager.GetPolicy(ctx, req, policyID)
+	}
 	if err != nil {
 		return nil, nil, goa.ErrInternal(err.Error())
 	}
 	return policy, pat, nil
 }
+
+// Creates Policy if missing. Can happen for old spaces.
+func (c *CollaboratorsController) createPolicy(ctx collaboratorContext, req *goa.RequestData, spc space.Space) (*auth.KeycloakPolicy, *string, error) {
+	resource, err := c.policyManager.CreateResource(ctx, req, spc.ID.String(), spaceResourceType, &spc.Name, &scopes, spc.OwnerId.String(), fmt.Sprintf("%s-%s", spc.Name, uuid.NewV4().String()))
+	if err != nil {
+		return nil, nil, err
+	}
+	spaceResource := &space.Resource{
+		ResourceID:   resource.ResourceID,
+		PolicyID:     resource.PolicyID,
+		PermissionID: resource.PermissionID,
+		SpaceID:      spc.ID,
+	}
+	err = application.Transactional(c.db, func(appl application.Application) error {
+		// Create space resource which will represent the keyclok resource associated with this space
+		_, err = appl.SpaceResources().Create(ctx, spaceResource)
+		return err
+	})
+	if err != nil {
+		// Clean up KC resource if transaction failed
+		resErr := c.policyManager.DeleteResource(ctx, req, *resource)
+		if resErr != nil {
+			return nil, nil, fmt.Errorf("%s %s", err.Error(), resErr.Error())
```

Comment:
Shouldn't we improve the error message ? To be able to distinguish which error is which... I believe if we get two traces that won't that easy.